docs/run-instance/installation/building-from-source.md

```diff
@@ -0,0 +1,153 @@
+# Building from Source
+
+If you need to customize your openrouteservice instance even further than what is possible by [configuration](/run-instance/configuration/), or want to start contributing to the openrouteservice project, the following section will give you starting points.
+
+## Prerequisites
+
+The following documentation assumes you are running an Ubuntu 20.04 system (also generally works with newer Ubuntu versions). Depending on your environent, you might need to adjust certain details. You will also need to make sure to have the following installed: 
+* [git](https://github.com/git-guides/install-git) should be available on your system
+* [java](https://www.java.com/en/) 17 should be available, preferably as default Java environment
+* [maven](https://maven.apache.org/) should be installed on your system
+
+## Installation from source
+
+To build openrouteservice from source, you can use the following commands:
+```shell
+git clone https://github.com/user/openrouteservice.git
+cd openrouteservice
+mvn package
+```
+
+If you have made modification to the code, you should run all tests before building and using openrouteservice. For a significant part of the tests you need to activate a maven profile called `apitests` to run them. Use the following commands:
+
+```shell
+mvn -Papitests verify
+# afterwards you can build the artefacts without running the tests again by issueing the folliwing command:
+mvn -DskipTests package
+```
+
+[//]: # (TODO: overhaul contents below after integrating the jar build PR)
+After running this command, you will find the artefact at `ors-api/target/ors.war`. 
+
+After you have packaged openrouteservice, there are two options for running it.
+One is to run the `mvn spring-boot:run` command which triggers a spring-boot native
```

Comment:
I've tried following the instructions but run into an error upon `mvn spring-boot:run`. Setting the env variable `export ORS_CONFIG_LOCATION=ors-api/ors-config.yml` doesn't resolve the issue.
```
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   No profiles configured. Exiting.
24 Jan 10:35:41 ERROR                                             ORS-Init [ o.h.o.r.RoutingProfileManager            ]   
```